clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -42,6 +42,8 @@ static bool shouldCacheInvocation(ArrayRef<const char *> Args,
       CheckArgs.end());
   CreateInvocationOptions Opts;
   Opts.Diags = Diags;
+  // This is useful to get a result for multi-arch invocations.
+  Opts.RecoverOnError = true;
```

Comment:
What is this actually doing that we need in the case of multi-arch? It seems like we shouldn't have errors.

Reply:
Getting more than one command job is treated as an error: https://github.com/apple/llvm-project/blob/next/clang/lib/Frontend/CreateInvocationFromCommandLine.cpp#L85

Reply:
I guess this code is only being used in `shouldCacheInvocation`, which is why it's okay to only handle a single invocation here - maybe we could update the comment to say "pick the first invocation in multi-arch builds"? Right now this is okay because the two options you're checking below are not arch-specific.

Reply:
Done 👍